test/base/MockCustomsCacheService.scala

```diff
@@ -17,18 +17,19 @@
 package base
 
 import forms.Choice
-import org.mockito.ArgumentMatchers
+import org.mockito.{ArgumentMatchers, Mockito}
 import org.mockito.ArgumentMatchers.any
 import org.mockito.Mockito.when
 import org.mockito.stubbing.OngoingStubbing
+import org.scalatest.{BeforeAndAfterEach, Suite}
 import org.scalatest.mockito.MockitoSugar
 import play.api.data.Form
 import services.CustomsCacheService
 import uk.gov.hmrc.http.cache.client.CacheMap
 
 import scala.concurrent.Future
 
-trait MockCustomsCacheService extends MockitoSugar {
+trait MockCustomsCacheService extends MockitoSugar with BeforeAndAfterEach { self: Suite =>
```

Comment:
What's the benefit of the `self: Suite => `?

Reply:
It is required by `BeforeAndAfterEach`